src/listen/websocket.rs

```diff
@@ -37,13 +40,9 @@ use crate::{Deepgram, DeepgramError, Result, Transcription};
 static LIVE_LISTEN_URL_PATH: &str = "v1/listen";
 
 #[derive(Debug)]
-pub struct StreamRequestBuilder<'a, S, E>
-where
-    S: Stream<Item = std::result::Result<Bytes, E>>,
-{
+pub struct StreamRequestBuilder<'a> {
```

Comment:
The changes in this file move the generics off of the `StreamRequestBuilder`, and onto `StreamRequest`.  

Previously, `StreamRequestBuilder` carried a `stream: Option<S>`, which was None until the Builder was initialized with a stream.  But the type declaration forced the user to effectively "choose" a stream type when the builder is first created.  In order to support using the API without a stream (manually driving request and response messages in a loop), we actually want a case where there's never a stream.  So now the builder doesn't require that type, and when you choose to add a stream to it, you get a new type (`StreamRequest`) which contains a non-optional `stream: S`.  

In addition to unlocking the non-streaming case, this prevents misuse where a user might try to start the stream before it is initialized.

Reply:
Took me a few reads, but I think I follow. I assume we'll use the same pattern for TTS WS then, when that comes up.